### PR TITLE
add (hidden) option to configure smb timeout

### PR DESCRIPTION
**Pull request by icewind1991**
hidden from ui to prevent clutter

```occ files_external:config <mount> timeout 30```

Signed-off-by: Robin Appelman <robin@icewind.nl>

### CODE REVIEW

**Comment by icewind1991** [comment]
For the future we should look into providing an "advanced settings" mechanism for external storage to put rarely used options like this into

File: apps/files_external/lib/Lib/Backend/SMB.php [hunk 1]
```diff
@@ -53,6 +53,8 @@ public function __construct(IL10N $l, Password $legacyAuth) {
 				(new DefinitionParameter('show_hidden', $l->t('Show hidden files')))
 					->setType(DefinitionParameter::VALUE_BOOLEAN)
 					->setFlag(DefinitionParameter::FLAG_OPTIONAL),
+				(new DefinitionParameter('timeout', $l->t('Timeout')))
+					->setFlag(DefinitionParameter::VALUE_HIDDEN)
```
Comment (kesselb) [inline apps/files_external/lib/Lib/Backend/SMB.php]:
```suggestion
					->setType(DefinitionParameter::VALUE_HIDDEN)
```

Guessing from https://github.com/nextcloud/server/blob/c43e0d6e283773b5c531d55fbd652879849b9f4d/apps/files_external/templates/settings.php#L73 VALUE_HIDDEN should be the type.

Comment (icewind1991) [reply apps/files_external/lib/Lib/Backend/SMB.php]: :see_no_evil:

**Review by rullzer — APPROVED** [review]
Sure

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**